### ISSUE TITLE
optim.method = "gen" for DiceKriging::km was a bad choice (seed not respected)

### ISSUE THREAD

**Issue by jakob-r** [mlr-org/mlrMBO#404]
Can we roll back and use `optim.method = "BFGS"`? We may have light evidence that `rgenoud` creates fewer ill fits but we have many issues complaining about non reproducibility that are all due to `rgenoud`.
I already wrote the Maintainer of `rgenoud` and convinced him to use the R RNG but I don't see this change coming. 

So we can

1. write a fix for rgenoud (should be really easy) and ask them to pull it in
2. if this does not work upload a new package rgenoud_fix to CRAN??? and ask DiceKriging devs to use that?
3. Ask DiceKriging devs to call `genoud()` with the parameters `int.seed` and `unif.seed` set to a random number set by the r rng. [This random number however should not accidently match with the defaults because then they will be ignored and the optimizer will be initialized with a random number from C](https://github.com/cran/rgenoud/blob/800d9fdafe204076b0284b91a9094467277e0320/R/rgenoud.R#L227)
4. Switch back to BFGS

Related Issues:
- https://github.com/mlr-org/mlrMBO/issues/359 (please switch to BFGS because of stochastic errors)
- https://github.com/mlr-org/mlrMBO/issues/98 (please switch to BFGS)
- #402 (reproducibility)
- https://github.com/mlr-org/mlrMBO/issues/258 (performance comparision, BFGS -> more often constant models, end performance nearly similar to rgenoud)
- ... (maybe)

**Comment by berndbischl** [comment]
> Can we roll back and use optim.method = "BFGS"

we have a paper now on arvix where we benchmark our current defaults. so such a rollback is not simple.



> I already wrote the Maintainer of rgenoud and convinced him to use the R RNG but I don't see this change coming.

so you didnt convince him.....

> write a fix for rgenoud (should be really easy) and ask them to pull it in

why not try that first? if it is really easy?

> f this does not work upload a new package rgenoud_fix to CRAN??? and ask DiceKriging devs to use that?

... no ...

**Comment by berndbischl** [comment]
> #359 (please switch to BFGS because of stochastic errors)

i replied there. and the issue never says "please switch"

**Comment by berndbischl** [comment]
all of the other issues are not truely about problems with rgenoud, but concern different things.

I have no problem talking about such a switch. but you also need to make a proposal how this is communicated wrt to the paper.

**Comment by jakob-r** [comment]
> we have a paper now on arvix where we benchmark our current defaults. so such a rollback is not simple.
> I have no problem talking about such a switch. but you also need to make a proposal how this is communicated wrt to the paper.

The paper would be about mlrMBO v1.1x. But of course if we want to submit it further it would be strange to have it be based on an old version. At another hand... thanks to this issue the results won't be reproducible anyhow.

> so you didnt convince him.....

I did but apart from the words _"I think the right thing to do is to just switch to using R RNG."_ nothing changed since the end of june. I emailed Jaseet again yesterday and he told me that the current code is beeing tested and that it should be fixed very soon.
